src/components/Modal/__stories__/modal.stories.mdx

```diff
@@ -0,0 +1,308 @@
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+              <TextField title="Email" size={TextField.sizes.MEDIUM} placeholder="email@monday.com" />
+            </form>
+            <ModalFooter>{modalButtons}</ModalFooter>
+          </Modal>
+        </div>
+      );
+    }}
+  </Story>
+</Canvas>
+
+### Modal with custom Header
+
+Header with description and icon
+
+<Canvas>
+  <Story name="Modal with custom Header">
+    {() => {
+      const { openModalButton, modalButtons, show, closeModal, ref } = useModalHelper({
+        openButtonTitle: "Default"
+      });
+      return (
+        <div>
+          {openModalButton}
+          <Modal
+            id={"story-book-modal"}
+            title={"Modal title"}
+            triggerElement={ref.current}
+            show={show}
+            onClose={closeModal}
+            closeButtonAriaLabel={"close"}
+          >
+            <ModalHeader title={"Modal Title"} description={"Modal description goes here"} icon={Settings} />
+            <div>Modal content goes here</div>
+            <ModalFooter>{modalButtons}</ModalFooter>
+          </Modal>
+        </div>
+      );
+    }}
+  </Story>
+</Canvas>
+
+## Related components
```

Comment:
I think we can add:
Popover, Tooltip, Alert as related components.

And what about Do's and Don'ts ?

Reply:
I agree. Not sure we have vibe Popover component, but we do have dialog container component.

Reply:
my suggestion: TOOLTIP, DIALOG_CONTAINER, Tipseen.